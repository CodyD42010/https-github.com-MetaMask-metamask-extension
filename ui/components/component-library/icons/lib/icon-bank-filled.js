import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconBankFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M456 399v62H47v-62c0-11 9-20 20-20h369c11 0 20 9 20 20zM149 236h-41v143h41zm82 0h-41v143h41zm82 0h-41v143h41zm82 0h-41v143h41zm82 240H26c-8 0-15-7-15-15 0-9 7-16 15-16h451c8 0 15 7 15 16 0 8-7 15-15 15zm-34-348L259 54c-4-1-11-1-15 0L60 128c-8 3-13 11-13 19v68c0 11 9 21 20 21h369c11 0 20-10 20-21v-68c0-8-5-16-13-19zm-191 56c-17 0-31-13-31-30s14-31 31-31 30 14 30 31-13 30-30 30z" />,
  );
};
IconBankFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
