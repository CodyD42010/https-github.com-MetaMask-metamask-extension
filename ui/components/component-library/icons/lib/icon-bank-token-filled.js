import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconBankTokenFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M22 15c0 4-3 7-7 7l1-2M2 9c0-4 3-7 7-7L8 4m223 451H26v-31c0-5 5-10 10-10h185c5 0 10 5 10 10zm-6-166-93-37c-2-1-5-1-7 0l-92 37c-4 1-7 5-7 9v34c0 6 5 11 10 11h185c5 0 10-5 10-11v-34c0-4-3-8-6-9zm-76 8c0 11-9 20-20 20-12 0-21-9-21-20s9-21 21-21c11 0 20 10 20 21zm82 158v8c4 0 8-4 8-8zm-205 0h-7c0 4 3 8 7 8zm13-41c0 4 3 8 8 8 4 0 7-4 7-8zm15-71c0-5-3-8-7-8-5 0-8 3-8 8zm26 71c0 4 3 8 8 8 4 0 7-4 7-8zm15-71c0-5-3-8-7-8-5 0-8 3-8 8zm26 71c0 4 3 8 8 8 4 0 7-4 7-8zm15-71c0-5-3-8-7-8-5 0-8 3-8 8zm26 71c0 4 3 8 8 8 4 0 7-4 7-8zm15-71c0-5-3-8-7-8-5 0-8 3-8 8zm26 71c0 4 3 8 8 8 4 0 7-4 7-8zm15-71c0-5-3-8-7-8-5 0-8 3-8 8zM16 447c-4 0-8 4-8 8s4 8 8 8zm225 16c4 0 8-4 8-8s-4-8-8-8zM132 252l-2 7zm93 37 2-7zm-100-37 3 7zm-92 37-3-7zm198 158H26v16h205zm-197 8v-31H19v31zm0-31c0-1 1-2 2-2v-15c-9 0-17 8-17 17zm2-2h185v-15H36zm185 0c1 0 2 1 2 2h16c0-9-8-17-18-17zm2 2v31h16v-31zM54 414v-71H39v71zm41 0v-71H80v71zm41 0v-71h-15v71zm41 0v-71h-15v71zm41 0v-71h-15v71zM16 463h225v-16H16zm114-204 92 37 5-14-92-37zm-2 0h2l5-14c-2-1-4-1-6-1s-5 0-7 1zm-92 37 92-37-6-14-92 37zm-2 2s0-1 1-1v-1h1l-6-14c-4 1-6 4-8 6-2 3-3 7-3 10zm0 34v-34H19v34zm2 3c-1 0-2-1-2-3H19c0 10 8 18 17 18zm185 0H36v15h185zm2-3c0 2-1 3-2 3v15c10 0 18-8 18-18zm0-34v34h16v-34zm-1-2 1 1v1h16c0-3-2-7-3-10-2-2-5-5-9-6zm-93 29c15 0 28-13 28-28h-16c0 7-5 13-12 13zm-29-28c0 15 13 28 29 28v-15c-7 0-13-6-13-13zm29-28c-16 0-29 12-29 28h16c0-7 6-13 13-13zm28 28c0-16-13-28-28-28v15c7 0 12 6 12 13zm107-163 65-28c3-2 6-2 10 0l65 28c6 3 11-4 7-9l-68-83c-4-6-12-6-16 0l-69 83c-5 5 0 12 6 9zm0 80 65 28c3 2 6 2 10 0l65-28c6-3 11 4 7 9l-68 84c-4 5-12 5-16 0l-69-84c-5-5 0-12 6-9zm69-75-71 35 71 36 71-36z" />,
  );
};
IconBankTokenFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
