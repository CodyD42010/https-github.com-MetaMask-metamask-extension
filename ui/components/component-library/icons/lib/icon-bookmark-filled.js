import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconBookmarkFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M456 170v172c0 66-53 119-119 119H166c-66 0-119-53-119-119V170c0-51 21-88 58-106 13-6 29 4 29 19v182c0 24 9 43 26 53s39 8 61-5l27-16c1-1 6-1 7-1l27 17c15 8 27 11 38 11 10 0 18-3 23-6 17-10 26-29 26-53V83c0-15 16-25 29-19 37 18 58 55 58 106zM318 51c11 0 21 9 21 21v193c0 13-4 23-11 26-7 5-18 2-30-5l-27-16c-11-6-28-6-39 0l-27 16c-12 7-23 9-30 5-7-3-11-13-11-26V72c0-12 10-21 21-21z" />,
  );
};
IconBookmarkFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
