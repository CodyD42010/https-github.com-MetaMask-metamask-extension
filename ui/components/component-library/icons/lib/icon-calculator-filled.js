import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconCalculatorFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M333 51H170C113 51 67 97 67 154v204c0 57 46 103 103 103h163c57 0 103-46 103-103V154c0-57-46-103-103-103zM187 392c-4 4-9 6-14 6-6 0-11-2-15-6s-6-9-6-15c0-5 2-10 6-14 2-2 4-4 7-5 5-2 11-2 16 0 1 1 2 1 3 2l3 3c4 4 6 9 6 14 0 6-2 11-6 15zm-35-97c0-2 1-5 2-7 1-3 2-5 4-7 5-5 12-7 19-6 1 0 2 1 4 1 1 1 2 2 3 2l3 3c2 2 4 4 5 7 1 2 1 5 1 7 0 6-2 11-6 15s-9 6-14 6c-3 0-6-1-8-2-3-1-5-2-7-4-4-4-6-9-6-15zm117 97c-2 2-4 3-6 4-3 1-6 2-8 2-6 0-11-2-15-6s-6-9-6-15c0-1 0-2 1-4 0-1 0-2 1-4 0-1 1-2 2-3 0-1 1-2 2-3 2-2 4-4 7-5 7-3 16-1 22 5 4 4 6 9 6 14 0 6-2 11-6 15zm0-82c-4 4-9 6-14 6-6 0-11-2-15-6s-6-9-6-15c0-5 2-10 6-14 8-8 22-8 29 0 2 2 4 4 5 7 1 2 1 5 1 7 0 6-2 11-6 15zm-79-86c-21 0-38-17-38-38v-21c0-21 17-38 38-38h123c21 0 38 17 38 38v21c0 21-17 38-38 38zm161 168c-4 4-9 6-14 6-3 0-6-1-8-2-3-1-5-2-7-4-4-4-6-9-6-15 0-5 2-10 6-14 6-6 15-8 23-5 2 1 4 3 6 5 4 4 6 9 6 14 0 6-2 11-6 15zm5-89c-1 3-3 5-5 7-4 4-9 6-14 6-6 0-11-2-15-6s-6-9-6-15c0-5 2-10 6-14 8-8 21-8 29 0 4 4 6 9 6 14 0 3 0 6-1 8z" />,
  );
};
IconCalculatorFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
