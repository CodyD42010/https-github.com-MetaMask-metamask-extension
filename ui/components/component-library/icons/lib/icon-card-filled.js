import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconCardFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M456 165c0 13-11 24-24 24H71c-13 0-24-11-24-24 0-47 38-85 84-85h240c47 0 85 38 85 85zM47 245v102c0 47 38 85 84 85h240c47 0 85-38 85-85V245c0-14-11-25-24-25H71c-13 0-24 11-24 25zm123 119h-41c-9 0-16-7-16-16 0-8 7-15 16-15h41c8 0 15 7 15 15 0 9-7 16-15 16zm133 0h-82c-9 0-16-7-16-16 0-8 7-15 16-15h82c8 0 15 7 15 15 0 9-7 16-15 16z" />,
  );
};
IconCardFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
