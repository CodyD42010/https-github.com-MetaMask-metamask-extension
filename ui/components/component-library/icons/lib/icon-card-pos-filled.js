import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconCardPosFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M456 445c0 9-7 16-15 16H62c-8 0-15-7-15-16 0-8 7-15 15-15h379c8 0 15 7 15 15zM321 103 101 323c-8 8-22 8-30 0-29-29-29-75 0-104L217 73c29-29 75-29 104 0 8 8 8 22 0 30zm111 81-62-62c-9-9-22-9-30 0L120 342c-9 8-9 21 0 30l62 62c29 29 75 29 104 0l146-146c29-29 29-75 0-104zM267 369l-25 25c-5 5-13 5-19 0-5-5-5-13 0-19l25-25c5-5 14-5 19 0s5 14 0 19zm81-81-50 50c-5 5-13 5-18 0-6-5-6-14 0-19l50-50c5-5 13-5 18 0 6 5 6 14 0 19z" />,
  );
};
IconCardPosFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
