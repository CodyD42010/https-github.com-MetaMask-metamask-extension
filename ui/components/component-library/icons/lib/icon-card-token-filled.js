import * as React from 'react';
import PropTypes from 'prop-types';
import { SIZES } from '../../../../helpers/constants/design-system';
import Box from '../../../ui/box/box';
import { BaseIcon } from '../../base-icon';
export const IconCardTokenFilled = ({ size, color, className, ...props }) => {
  return React.createElement(
    BaseIcon,
    {
      size,
      color,
      className,
      ...props,
    },
    <path d="M252 332v2c0 6-5 10-11 10H57c-6 0-10-4-10-10v-2c0-36 9-45 45-45h114c37 0 46 9 46 45zM57 365c-6 0-10 4-10 10v41c0 36 9 45 45 45h114c37 0 46-9 46-45v-41c0-6-5-10-11-10zM22 15c0 4-3 7-7 7l1-2M2 9c0-4 3-7 7-7L8 4m256 130 65-28c3-2 6-2 9 0l65 28c6 3 11-4 7-9l-68-83c-5-6-13-6-17 0l-68 83c-4 5 1 12 7 9zm0 80 65 29c3 1 6 1 9 0l65-29c6-3 11 4 7 9l-68 83c-5 6-12 6-17 0l-68-83c-4-5 1-12 7-9zm66-74-56 28c-5 2-5 10 0 12l56 28c2 1 5 1 7 0l56-28c5-2 5-10 0-12l-56-28c-2-1-5-1-7 0z" />,
  );
};
IconCardTokenFilled.propTypes = {
  /**
   * The size of the BaseIcon.
   * Possible values could be 'xxs', 'xs', 'sm', 'md', 'lg', 'xl',
   */
  size: PropTypes.oneOf(Object.values(SIZES)),

  /**
   * The color of the icon. Defaults to 'inherit'.
   */
  color: PropTypes.string,

  /**
   * An additional class name to apply to the icon.
   */
  className: PropTypes.string,

  /**
   * BaseIcon accepts all the props from Box
   */
  ...Box.propTypes,
};
