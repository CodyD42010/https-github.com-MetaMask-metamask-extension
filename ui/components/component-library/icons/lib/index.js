export { IconAddSquareFilled } from './icon-add-square-filled';
export { IconBankFilled } from './icon-bank-filled';
export { IconBankTokenFilled } from './icon-bank-token-filled';
export { IconBookmarkFilled } from './icon-bookmark-filled';
export { IconCalculatorFilled } from './icon-calculator-filled';
export { IconCardFilled } from './icon-card-filled';
export { IconCardPosFilled } from './icon-card-pos-filled';
export { IconCardTokenFilled } from './icon-card-token-filled';
export { IconCategoryFilled } from './icon-category-filled';
export { IconChartFilled1 } from './icon-chart-filled-1';
export { IconChartFilled } from './icon-chart-filled';
export { IconCheckBoxOffOutline } from './icon-check-box-off-outline';
export { IconCheckBoxOnFilled } from './icon-check-box-on-filled';
export { IconCheckCircleOnFilled } from './icon-check-circle-on-filled';
export { IconClockFilled } from './icon-clock-filled';
export { IconCloseOutline } from './icon-close-outline';
export { IconCodeCircleFilled } from './icon-code-circle-filled';
export { IconCoinFilled } from './icon-coin-filled';
export { IconConnectFilled } from './icon-connect-filled';
export { IconCopyFilled } from './icon-copy-filled';
export { IconCopySuccessFilled } from './icon-copy-success-filled';
export { IconDangerFilled } from './icon-danger-filled';
export { IconDarkFilled } from './icon-dark-filled';
export { IconDataFilled } from './icon-data-filled';
export { IconDocumentCodeFilled } from './icon-document-code-filled';
export { IconEditFilled } from './icon-edit-filled';
export { IconEraserFilled } from './icon-eraser-filled';
export { IconExploreFilled } from './icon-explore-filled';
export { IconEyeFilled } from './icon-eye-filled';
export { IconEyeSlashFilled } from './icon-eye-slash-filled';
export { IconFilterOutline } from './icon-filter-outline';
export { IconFlagFilled } from './icon-flag-filled';
export { IconFlashFilled } from './icon-flash-filled';
export { IconFlashSlashFilled } from './icon-flash-slash-filled';
export { IconFullCircleFilled } from './icon-full-circle-filled';
export { IconGasFilled } from './icon-gas-filled';
export { IconGlobalFilled } from './icon-global-filled';
export { IconGlobalSearchFilled } from './icon-global-search-filled';
export { IconGraphFilled } from './icon-graph-filled';
export { IconHeartFilled } from './icon-heart-filled';
export { IconHierarchyFilled } from './icon-hierarchy-filled';
export { IconHomeFilled } from './icon-home-filled';
export { IconInfoFilled } from './icon-info-filled';
export { IconKeyFilled } from './icon-key-filled';
export { IconLightFilled } from './icon-light-filled';
export { IconLink2Outline } from './icon-link-2-outline';
export { IconLoadingFilled } from './icon-loading-filled';
export { IconLockCircleFilled } from './icon-lock-circle-filled';
export { IconLockFilled } from './icon-lock-filled';
export { IconLockSlashFilled } from './icon-lock-slash-filled';
export { IconMenuOutline } from './icon-menu-outline';
export { IconMessageQuestionFilled } from './icon-message-question-filled';
export { IconMessagesFilled } from './icon-messages-filled';
export { IconMinusSquareFilled } from './icon-minus-square-filled';
export { IconMobileFilled } from './icon-mobile-filled';
export { IconMoneyFilled } from './icon-money-filled';
export { IconMonitorFilled } from './icon-monitor-filled';
export { IconMoreHorizontalOutline } from './icon-more-horizontal-outline';
export { IconMoreVerticalOutline } from './icon-more-vertical-outline';
export { IconNotificationCircleFilled } from './icon-notification-circle-filled';
export { IconNotificationFilled } from './icon-notification-filled';
export { IconPasswordCheckFilled } from './icon-password-check-filled';
export { IconPeopleFilled } from './icon-people-filled';
export { IconProgrammingArrowsFilled } from './icon-programming-arrows-filled';
export { IconQuestionFilled } from './icon-question-filled';
export { IconReceivedOutline } from './icon-received-outline';
export { IconScanBarcodeFilled } from './icon-scan-barcode-filled';
export { IconScanFilled } from './icon-scan-filled';
export { IconScrollFilled } from './icon-scroll-filled';
export { IconSearchFilled } from './icon-search-filled';
export { IconSecurityCardFilled } from './icon-security-card-filled';
export { IconSecurityCrossFilled } from './icon-security-cross-filled';
export { IconSecurityKeyFilled } from './icon-security-key-filled';
export { IconSecuritySearchFilled } from './icon-security-search-filled';
export { IconSecuritySlashFilled } from './icon-security-slash-filled';
export { IconSecurityTickFilled } from './icon-security-tick-filled';
export { IconSecurityTimeFilled } from './icon-security-time-filled';
export { IconSecurityUserFilled } from './icon-security-user-filled';
export { IconSend1Filled } from './icon-send-1-filled';
export { IconSend2Outline } from './icon-send-2-outline';
export { IconSettingFilled } from './icon-setting-filled';
export { IconSlashFilled } from './icon-slash-filled';
export { IconSnapsFilled } from './icon-snaps-filled';
export { IconSnapsMobileFilled } from './icon-snaps-mobile-filled';
export { IconSnapsPlusFilled } from './icon-snaps-plus-filled';
export { IconSpeedometerFilled } from './icon-speedometer-filled';
export { IconStudentFilled } from './icon-student-filled';
export { IconSwapHorizontalOutline } from './icon-swap-horizontal-outline';
export { IconSwapVerticalFilled } from './icon-swap-vertical-filled';
export { IconTagFilled } from './icon-tag-filled';
export { IconTimerFilled } from './icon-timer-filled';
export { IconTrashFilled } from './icon-trash-filled';
export { IconTrendDownFilled } from './icon-trend-down-filled';
export { IconTrendUpFilled } from './icon-trend-up-filled';
export { IconUserAddFilled1 } from './icon-user-add-filled-1';
export { IconUserAddFilled } from './icon-user-add-filled';
export { IconUserCircleAddFilled } from './icon-user-circle-add-filled';
export { IconUserCircleFilled } from './icon-user-circle-filled';
export { IconUserFilled } from './icon-user-filled';
export { IconUserMinusFilled } from './icon-user-minus-filled';
export { IconUserRemoveFilled } from './icon-user-remove-filled';
export { IconUserSearchFilled } from './icon-user-search-filled';
export { IconUserTickFilled } from './icon-user-tick-filled';
export { IconWalletCardFilled } from './icon-wallet-card-filled';
export { IconWalletFilled } from './icon-wallet-filled';
export { IconWalletMoneyFilled } from './icon-wallet-money-filled';
export { IconWarningFilled } from './icon-warning-filled';
