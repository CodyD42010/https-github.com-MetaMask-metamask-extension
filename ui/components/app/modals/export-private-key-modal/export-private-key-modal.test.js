import React from 'react';
import { fireEvent } from '@testing-library/react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { renderWithProvider } from '../../../../../test/lib/render-helpers';
import mockState from '../../../../../test/data/mock-state.json';
import * as actions from '../../../../store/actions';
import ExportPrivateKeyModal from '.';

jest.mock('../../../../store/actions.ts', () => ({
  ...jest.requireActual('../../../../store/actions.ts'),
  exportAccount: jest.fn().mockReturnValue(jest.fn().mockResolvedValueOnce()),
  hideWarning: () => jest.fn(),
  showModal: () => jest.fn(),
  hideModal: () => jest.fn(),
  clearAccountDetails: () => jest.fn(),
}));

describe('Export PrivateKey Modal', () => {
  const password = 'a-password';

  const privKeyModalState = {
    ...mockState,
    appState: {
      ...mockState.appState,
      accountDetail: {
        privateKey: '0xPrivKey',
      },
    },
  };
  const mockStore = configureMockStore([thunk])(privKeyModalState);

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const { container } = renderWithProvider(
      <ExportPrivateKeyModal />,
      mockStore,
    );
    expect(container).toMatchSnapshot();
  });

  it('should disable confirm button by default', () => {
    const { queryByText } = renderWithProvider(
      <ExportPrivateKeyModal />,
      mockStore,
    );

    const confirmButton = queryByText('Confirm');

    expect(confirmButton).toBeDisabled();
  });

  it('should call export account with password and selected address', () => {
    const { queryByTestId, queryByText } = renderWithProvider(
      <ExportPrivateKeyModal />,
      mockStore,
    );

    const passwordInput =
      queryByTestId('password-input').querySelector('input');

    const passwordInputEvent = {
      target: {
        value: password,
      },
    };

    fireEvent.change(passwordInput, passwordInputEvent);

    const confirmButton = queryByText('Confirm');

    fireEvent.click(confirmButton);

    expect(actions.exportAccount).toHaveBeenCalledWith(
      password,
      mockState.metamask.selectedAddress,
    );
  });
});
