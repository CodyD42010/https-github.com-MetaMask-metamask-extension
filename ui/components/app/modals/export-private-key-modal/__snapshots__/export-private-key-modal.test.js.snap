// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Export PrivateKey Modal should match snapshot 1`] = `
<div>
  <div
    class="export-private-key-modal account-modal"
    style="border-radius: 4px;"
  >
    <div
      class="account-modal__container"
    >
      <div>
        <div
          class=""
        >
          <div
            class="identicon"
            style="height: 64px; width: 64px; border-radius: 32px;"
          >
            <div
              style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 64px; height: 64px; display: inline-block; background: rgb(250, 58, 0);"
            >
              <svg
                height="64"
                width="64"
                x="0"
                y="0"
              >
                <rect
                  fill="#18CDF2"
                  height="64"
                  transform="translate(-2.09678700758788 -6.608568138920997) rotate(328.9 32 32)"
                  width="64"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#035E56"
                  height="64"
                  transform="translate(-36.59692341766409 21.1849237434972) rotate(176.2 32 32)"
                  width="64"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#F26602"
                  height="64"
                  transform="translate(33.335684036447844 -28.410279445994163) rotate(468.9 32 32)"
                  width="64"
                  x="0"
                  y="0"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <button
        class="account-modal__close"
      />
      <span
        class="box mm-text mm-text--body-lg-medium mm-text--font-weight-normal box--margin-top-2 box--flex-direction-row box--color-text-default"
      >
        Test Account
      </span>
      <div
        class="box ellip-address-wrapper box--margin-top-2 box--padding-1 box--sm:padding-2 box--md:padding-1 box--lg:padding-2 box--flex-direction-row box--border-style-solid box--border-color-border-default box--border-width-1"
      >
        0x0DCD5D886577d5081B0c52e242Ef29E70Be3E7bc
      </div>
      <div
        class="box export-private-key-modal__divider box--margin-5 box--md:margin-3 box--flex-direction-row box--width-full"
      />
      <p
        class="box mm-text mm-text--body-lg-medium mm-text--font-weight-normal box--margin-4 box--md:margin-4 box--flex-direction-row box--color-text-default"
      >
        Show Private Keys
      </p>
      <div
        class="box box--padding-right-5 box--padding-left-5 box--display-flex box--flex-direction-column box--align-items-flex-start box--width-full"
      >
        <label
          class="box mm-text mm-label mm-text--body-sm mm-text--font-weight-bold box--margin-bottom-2 box--display-inline-flex box--flex-direction-row box--align-items-center box--color-text-default"
        >
          Type your MetaMask password
        </label>
        <div
          class="box mm-text-field mm-text-field--size-md mm-text-field--truncate export-private-key-modal__password-input box--display-inline-flex box--flex-direction-row box--align-items-center box--width-full box--background-color-background-default box--rounded-sm box--border-width-1 box--border-style-solid"
          data-testid="password-input"
        >
          <input
            autocomplete="off"
            class="box mm-text mm-input mm-input--disable-state-styles mm-text-field__input mm-text--body-md box--padding-right-4 box--padding-left-4 box--flex-direction-row box--color-text-default box--background-color-transparent box--border-style-none"
            focused="false"
            placeholder="Enter password"
            type="password"
            value=""
          />
        </div>
      </div>
      <p
        class="box mm-text mm-text--body-sm box--flex-direction-row box--color-error-default"
      />
      <div
        class="box mm-banner-base mm-banner-alert mm-banner-alert--severity-danger box--margin-top-4 box--margin-right-5 box--margin-left-5 box--padding-1 box--sm:padding-3 box--lg:padding-3 box--padding-left-2 box--display-flex box--gap-2 box--flex-direction-row box--background-color-error-muted box--rounded-sm"
      >
        <span
          class="box mm-icon mm-icon--size-lg box--display-inline-block box--flex-direction-row box--color-error-default"
          style="mask-image: url('./images/icons/danger.svg');"
        />
        <div>
          <p
            class="box mm-text mm-text--body-md box--flex-direction-row box--color-text-default"
          >
            Warning: Never disclose this key. Anyone with your private keys can steal any assets held in your account.
          </p>
        </div>
      </div>
      <div
        class="box box--margin-top-3 box--padding-5 box--md:padding-5 box--display-flex box--flex-direction-row box--justify-content-space-between box--width-full"
      >
        <button
          class="box mm-text mm-button-base mm-button-base--size-lg mm-button-primary mm-text--body-md box--margin-right-4 box--padding-right-4 box--padding-left-4 box--display-inline-flex box--flex-direction-row box--justify-content-center box--align-items-center box--width-1/2 box--color-primary-inverse box--background-color-primary-default box--rounded-pill"
          type="secondary"
        >
          Cancel
        </button>
        <button
          class="box mm-text mm-button-base mm-button-base--size-lg mm-button-base--disabled mm-button-primary mm-button-primary--disabled mm-text--body-md box--padding-right-4 box--padding-left-4 box--display-inline-flex box--flex-direction-row box--justify-content-center box--align-items-center box--width-1/2 box--color-primary-inverse box--background-color-primary-default box--rounded-pill"
          disabled=""
          type="primary"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
